app/controllers/users_controller.rb

```diff
@@ -4,10 +4,15 @@ def new
   end
 
   def create
-    user = User.create!(user_params)
-    session[:user_id] = user.id
-    flash[:success] = "Welcome #{user.name}! You are now registered and logged in."
-    redirect_to "/profile"
+    user = User.new(user_params)
+    if user.save
+      session[:user_id] = user.id
+      flash[:success] = "Welcome #{user.name}! You are now registered and logged in."
+      redirect_to "/profile"
+    else
+      flash[:error] = user.errors.full_messages.uniq.to_sentence
```

Comment:
Nice! I haven't used uniq in this before- what functionality does that add?

Reply:
I'm honestly not sure if we need it, but I threw it in there just to make sure we don't get duplicate flash error messages.